### PR TITLE
Fix NPE when method was called on an array type

### LINKED ISSUE #87562: Script: painless lookup npe for annotation
I got an unexpected internal NPE when playing with this script.

```
POST _scripts/painless/_execute
{
  "script": {
    "source": """
    return /[^\w]+/.split('abc   def higkln').toArray(String[]::new)
    """
  }
}
```
```
{
  "error" : {
    "root_cause" : [
      {
        "type" : "script_exception",
        "reason" : "compile error",
        "script_stack" : [ ],
        "script" : "\n    return /[^\\w]+/.split('abc   def higkln').toArray(String[]::new)\n    ",
        "lang" : "painless"
      }
    ],
    "type" : "script_exception",
    "reason" : "compile error",
    "script_stack" : [ ],
    "script" : "\n    return /[^\\w]+/.split('abc   def higkln').toArray(String[]::new)\n    ",
    "lang" : "painless",
    "caused_by" : {
      "type" : "null_pointer_exception",
      "reason" : "Cannot read field \"annotations\" because the return value of \"org.elasticsearch.painless.lookup.PainlessLookup.lookupPainlessClass(java.lang.Class)\" is null"
    }
  },
  "status" : 400
}

```

### PR DESCRIPTION
**Pull request by thecoop**
Fix non-obvious NPE when method was called on an array type

This fixes #87562

### CODE REVIEW

File: modules/lang-painless/src/test/java/org/elasticsearch/painless/WhenThingsGoWrongTests.java [hunk 1]
```diff
@@ -840,4 +840,8 @@ public void testInstanceMethodNotFound() {
         iae = expectScriptThrows(IllegalArgumentException.class, () -> exec("doesNotExist(1, 'string', false)"));
         assertEquals(iae.getMessage(), "Unknown call [doesNotExist] with [3] arguments.");
     }
+
+    public void testArrayToArrayException() {
+        expectScriptThrows(IllegalArgumentException.class, () -> exec("return new String[] {'a'}.method()"));
```
Comment (jdconrad) [inline modules/lang-painless/src/test/java/org/elasticsearch/painless/WhenThingsGoWrongTests.java:845]: Would you please add a snippet to compare the exception message? We throw `IllegalArgumentException` for a number of different errors, so I would like to make sure this is throwing the error we expect.

**Review by jdconrad — APPROVED** [review]
One minor comment.

File: modules/lang-painless/src/test/java/org/elasticsearch/painless/WhenThingsGoWrongTests.java (hunk 1 above)
Comment (thecoop) [reply modules/lang-painless/src/test/java/org/elasticsearch/painless/WhenThingsGoWrongTests.java]: Done

**Comment by thecoop** [comment]
@elasticsearchmachine rerun elasticsearch-ci/part-1

**Pull request merged**